### PR TITLE
Wip solution

### PR DESCRIPTION
**Pull request by alexanderGugel**

### CODE REVIEW

File: components/Sidebar.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+  }
+  handleClick(index, ev) {
+    ev.preventDefault();
+    this.props.onSelect(index)
+  }
+  render() {
+    const { files, selectedFileIndex, onAdd } = this.props;
```
Comment (thomastuts) [inline components/Sidebar.js]: `onAdd` isn't being used anywhere

File: components/Sidebar.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      <ul className="sidebar">
+        {
+          files.map((file, i) => (
+            <SidebarItem
+              isSelected={selectedFileIndex === i}
+              key={i}
+              file={file}
+              onClick={this.handleClick.bind(null, i)}
+            />
+          ))
```
Comment (thomastuts) [inline components/Sidebar.js:30]: Missing semicolon

File: components/App.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  }
+  handleSelect(selectedFileIndex) {
+    this.setState({ selectedFileIndex });
+  }
+  handleAdd(ev) {
+    ev.preventDefault();
+    actions.addFile();
+  }
+  handleRemove(ev) {
+    ev.preventDefault()
```
Comment (thomastuts) [inline components/App.js]: Missing semicolon

File: components/SidebarItem.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+const React = require('react');
+const classNames = require('classnames');
+
+class SidebarItem extends React.Component {
+  static getTitle (file) {
```
Comment (thomastuts) [inline components/SidebarItem.js]: While I like the use of a static method here, most students won't know what this is. I think we can extract this logic out into a function completely separate from the React component? (We can still keep it in the same file)

File: components/SidebarItem.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  render() {
+    const { file = '', isSelected, onClick } = this.props;
+    const title = SidebarItem.getTitle(file);
+
+    return (
+      <li className={classNames('sidebar__item', {
+        'sidebar__item--selected': isSelected
+      })}>
+        <a href='#' onClick={onClick} className='sidebar__link'>
+          {title ? title : <em>Untitled</em>}
```
Comment (thomastuts) [inline components/SidebarItem.js]: I think we can simplify this to `{title || <em>Untitled</em>}`

**Review by thomastuts — CHANGES_REQUESTED** [review]
Some minor code nits here, but nothing too big. Could we also add a line after every method in a class to improve code readability a little? I feel like the component code is pretty dense now.

File: components/Sidebar.js (hunk 2 above)
Comment (alexanderGugel) [reply components/Sidebar.js:30]: Hmmm..... Don't think so

**Comment by alexanderGugel** [comment]
@thomastuts Done! Ready to merge?

File: components/Sidebar.js (hunk 2 above)
Comment (thomastuts) [reply components/Sidebar.js:30]: Ah, sorry, you're absolutely right - this is an expression. Not sure how I missed that. Scratch that!

**Comment by thomastuts** [comment]
@alexanderGugel Yep! I'll let you pull the switch 😄

**Pull request merged**